src/main/java/baron/Models/Deadline.java

```diff
@@ -0,0 +1,46 @@
+package baron.Models;
+
+import baron.Utils.DateUtils;
+
+import java.time.LocalDateTime;
+
+public class Deadline extends Task {
+
+    private final LocalDateTime deadline;
+
+    public Deadline(String name, LocalDateTime deadline) {
+        super(name);
+        this.deadline = deadline;
+    }
+
+    public Deadline(int id, String name, boolean isDone, LocalDateTime deadline) {
+        super(id, name, isDone);
+        this.deadline = deadline;
+    }
```

Comment:
I personally like how you implemented method overloading but I feel that you could include Javadoc comments for constructors as well :)